db_stress_tool/multi_ops_txns_stress.cc

```diff
@@ -18,12 +18,116 @@
 
 namespace ROCKSDB_NAMESPACE {
 
-// TODO: move these to gflags.
-static constexpr uint32_t kInitNumC = 1000;
-#ifndef ROCKSDB_LITE
-static constexpr uint32_t kInitialCARatio = 3;
-#endif  // ROCKSDB_LITE
-static constexpr bool kDoPreload = true;
+DEFINE_int32(lb_a, 0, "(Inclusive) lower bound of A");
+DEFINE_int32(ub_a, 1000, "(Exclusive) upper bound of A");
+DEFINE_int32(lb_c, 0, "(Inclusive) lower bound of C");
+DEFINE_int32(ub_c, 1000, "(Exclusive) upper bound of C");
+
+DEFINE_string(key_spaces_path, "",
+              "Path to file describing the lower and upper bounds of A and C");
+
+DEFINE_int32(delay_snapshot_read_one_in, 0,
+             "With a chance of 1/N, inject a random delay between taking "
+             "snapshot and read.");
+
+// MultiOpsTxnsStressTest can either operate on a database with pre-populated
+// data (possibly from previous ones), or create a new db and preload it with
+// data specified via `-lb_a`, `-ub_a`, `-lb_c`, `-ub_c`, etc. Among these, we
+// define the test key spaces as two key ranges: [lb_a, ub_a) and [lb_c, ub_c).
+// The key spaces specification is persisted in a file whose absolute path can
+// be specified via `-key_spaces_path`.
```

Comment:
Key spaces specification are just 4 integers, right? Why do we want to persist it in a file instead of providing it from cmd?

Reply:
Right.
I feel that the key spaces should be persisted across database restarts if `destroy_db_initially` is false. This is different from the ranges of keys that actually exist in the database, thus we cannot rely on checking the db for this info.